File: _8_factorial.java
public class _8_factorial{

    static double factorial(int a){
        double ans = 1;
        for(int i=1 ; i<=a ; i++){
            ans = ans * i;
        }
        return ans;
    }

    static double factorial_recursion(int a){
        if(a==0){
            return 1;
        }
        return a * factorial_recursion(a-1); 
    }

    public static void main(String[] args){

        System.out.println(factorial(5));
        System.out.println(factorial_recursion(5));

    }
}

File: cpp/8_factorial.cpp
#include<iostream>
using namespace std;


unsigned long long int factorial(int a){
    unsigned long long int ans = 1;
    for(int i = 1 ; i<=a ; i++){
        ans = ans * i;
    }
    return ans;
}

double factorial_recursion(int a){
    if(a == 0){
        return 1;
    }
    return a * factorial_recursion(a-1);
}


int main(){

    cout<<factorial(13)<<endl;
    cout<<factorial_recursion(9)<<endl;
    

}